lib/index.js

```diff
@@ -367,14 +367,24 @@ class Config {
       if (!/^npm_config_/i.test(envKey) || envVal === '') {
         continue
       }
-      const key = envKey.slice('npm_config_'.length)
-        .replace(/(?!^)_/g, '-') // don't replace _ at the start of the key
-        .toLowerCase()
+      const key = this.normalizeKeyFromEnv(envKey.slice('npm_config_'.length))
       conf[key] = envVal
     }
     this[_loadObject](conf, 'env', 'environment')
   }
 
+  normalizeKeyFromEnv (key) {
+    // nerf-darted keys need slightly more nuanced handling
+    if (/^\/\/.*?:./.test(key)) {
+      // preserve case and dashes within URIs
+      let [uri, k] = key.split(':', 2)
+      k = this.normalizeKeyFromEnv(k)
+      return `${uri}:${k === '_authtoken' ? '_authToken' : k}`
+    }
```

Comment:
```suggestion
let [uri, k] = key.split(':')
if (uri.startsWith('//') && k) {
  // preserve case and dashes within URIs
  k = this.normalizeKeyFromEnv(k)
  return `${uri.slice(2)}:${k === '_authtoken' ? '_authToken' : k}`
}
```

This is maybe a 7/10 on the suggestion scale.  If you really think the regex is more appropriate here please push back.  We tend to default to not adding new regex (and removing existing regex) when possible, given the amount of bugs it's led to in npm the past.

Also, are we intentionally supporting `//uri:_authtoken` (lower case _authtoken)?  If we're not mucking with the case on uris do we still need to do this?

Reply:
disclaimer: I freehand typed that code, it may have linting/syntax errors. You may want to test and commit it locally.  I do not care about credit.

Reply:
After sleeping on this I realized that this will eat any urls with ports in them.  We probably need to take another stab at this so that doesn't happen.


```
npm_config_//my.server:4242/foo:_authToken=asdf
```

This isn't a problem we've had to solve yet because of the way we search through nerf-darted configs right now, (i.e. by url specifically).

`lastIndexOf` is likely the tool for the job.

```javascript
if (key.startsWith('//') && key.includes(':')) {
  // preserve case and dashes within URIs
  const uri = key.slice(2, key.lastIndexOf(':'))
  let k = key.slice(key.lastIndexOf(':') + 1)
  k = this.normalizeKeyFromEnv(k)
  return `${uri}:${k === '_authtoken' ? '_authToken' : k}`
}
```

Reply:
Sounds great, I've been offline for a couple days but will push a patch in the next day or so

Reply:
re: `//uri:_authtoken` on that line, it's moreso that I was maintaining the existing "key are case-insensitive" functionality where possible ... i.e. a nerfed key could have any case coming in (e.g. `//uri:_PASSWORD=...`), so we `normalizeKeyFromEnv` the sub-key on the next line. But then `_authtoken` then needs to be changed to `_authToken` since that's what various things downstream expect 😅

It's definitely a little weird, so if we're ok with saying that nerfed keys are case sensitive that could simplify this a bit... no need to normalize the sub-key or have special handling for `_authToken`. WDYT?

Reply:
one other note, we don't actually want to lop of the leading `//`, since all this function does is normalize casing/dashes where appropriate, we still want a fully nerfed config key including the slashes

Reply:
re case sensitivity: Windows is going to be the place to make sure things work as expected.  I know you can read from process.env ignoring case, but if you are iterating its keys, does it retain what was originally used?

Reply:
Yes, when iterating the keys in Windows you get the original value, e.g. under the `Output env` and `Output env 2` steps [here](https://github.com/jenseng/workflow-test/runs/7749300740?check_suite_focus=true) you can see the results of doing `for (let key in process.env) console.log(key)` under PowerShell and CMD.exe respectively, variable names like `mixed_CASE` and `something_//really-weird.LOL:_ok` are preserved

Reply:
Ok updated, it's much simpler now 😅

Reply:
That's an understatement!